### PR TITLE
add displayMode as  new param to sld (voltage-level) api

### PR DESCRIPTION
**Pull request by bendaamerahm**
add displayMode as new param to sld (voltage-level) api

### CODE REVIEW

**Review by geofjamg — CHANGES_REQUESTED** [review]
Instead of useFeederPositions we should have an enum so that it could be extended to other kind of display easily: 
something like displayMode:
 - StateVariable
 - FeederPosition

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -59,8 +59,7 @@ public StudyController(StudyService studyService,
             NetworkConversionService networkConversionService,
             SecurityAnalysisService securityAnalysisService,
             SensitivityAnalysisService sensitivityAnalysisService,
-            ShortCircuitService shortCircuitService,
-            CaseService caseService) {
+            ShortCircuitService shortCircuitService, CaseService caseService) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/study/server/StudyController.java:62]: undo

File: src/main/java/org/gridsuite/study/server/StudyConstants.java [hunk 2]
```diff
@@ -39,4 +39,9 @@ private StudyConstants() {
     public static final String QUERY_PARAM_ERROR_ON_GROUP_NOT_FOUND = "errorOnGroupNotFound";
     public static final String QUERY_PARAM_ERROR_ON_REPORT_NOT_FOUND = "errorOnReportNotFound";
     public static final String QUERY_PARAM_REPORT_DEFAULT_NAME = "defaultName";
+
+    public enum DisplayMode {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/study/server/StudyConstants.java:43]: Maybe use a more specific name like SldLabellingMode ? SldDisplayMode?

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyConstants.java (hunk 2 above)
Comment (bendaamerahm) [reply src/main/java/org/gridsuite/study/server/StudyConstants.java:43]: i agree

**Review by geofjamg — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**